### DISCUSSION TITLE
How can I run the same tests twice (or more) but with one thing different

### DISCUSSION THREAD

**Discussion by fregante** [avajs/ava#2800] [Q&A] [answered]
Let's say I have this

```js
const identity1 = x => x;
const identity2 = y => y; // unused

test('base', t => {
  test.is(identity1(555), 555)
});
```

I want to run this test a second time, but replacing part of it (`identity1` -> `identity2`). 

Is there any trick other than these 2 options?

- run ava twice: `ava && ID=2 ava`
- explicit loop:

	```js
	const identity1 = x => x;
	const identity2 = y => y;

	for (const id of [identity1, identity2]) {
		test('base ' + id.name, t => {
			test.is(id(555), 555)
		});
	}
	```

In this specific situation, I want to ensure that two implementations are identical, but I suppose it can be applied to anything else (testing multiple websites, multiple sources, etc)

**Comment by novemberborn** [comment, marked as answer]
I would use a macro:

```js
const checkIdentity = (t, fn) => {
  t.is(fn(555), 555)
}

checkIdentity.title = (title, fn) => `${title || ''} ${fn.name}`

test('base', checkIdentity, identity1)
test('base', checkIdentity, identity2)
```
